commit 7a311dca7708162fcc047282affe85381e5eabb8
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date:   Tue Dec 11 16:11:54 2018 -0300

    perf header: Fix up argument to ctime()
    
    [ Upstream commit 0afcf29bab35d3785204cd9bd51693b231ad7181 ]
    
    Reducing this noise when cross building to the Android NDK:
    
      util/header.c: In function 'perf_header__fprintf_info':
      util/header.c:2710:45: warning: pointer targets in passing argument 1 of 'ctime' differ in signedness [-Wpointer-sign]
        fprintf(fp, "# captured on    : %s", ctime(&st.st_ctime));
                                                   ^
      In file included from util/../perf.h:5:0,
                       from util/evlist.h:11,
                       from util/header.c:22:
      /opt/android-ndk-r15c/platforms/android-26/arch-arm/usr/include/time.h:81:14: note: expected 'const time_t *' but argument is of type 'long unsigned int *'
       extern char* ctime(const time_t*) __LIBC_ABI_PUBLIC__;
                    ^
    
    Cc: Adrian Hunter <adrian.hunter@intel.com>
    Cc: Jiri Olsa <jolsa@kernel.org>
    Cc: Namhyung Kim <namhyung@kernel.org>
    Link: https://lkml.kernel.org/n/tip-6bz74zp080yhmtiwb36enso9@git.kernel.org
    Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/tools/perf/util/header.c b/tools/perf/util/header.c
index ac3c59e7da16..bd9226bc5945 100644
--- a/tools/perf/util/header.c
+++ b/tools/perf/util/header.c
@@ -2636,6 +2636,7 @@ int perf_header__fprintf_info(struct perf_session *session, FILE *fp, bool full)
 	struct perf_header *header = &session->header;
 	int fd = perf_data__fd(session->data);
 	struct stat st;
+	time_t stctime;
 	int ret, bit;
 
 	hd.fp = fp;
@@ -2645,7 +2646,8 @@ int perf_header__fprintf_info(struct perf_session *session, FILE *fp, bool full)
 	if (ret == -1)
 		return -1;
 
-	fprintf(fp, "# captured on    : %s", ctime(&st.st_ctime));
+	stctime = st.st_ctime;
+	fprintf(fp, "# captured on    : %s", ctime(&stctime));
 
 	fprintf(fp, "# header version : %u\n", header->version);
 	fprintf(fp, "# data offset    : %" PRIu64 "\n", header->data_offset);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1162_7a311dca7708162fcc047282affe85381e5eabb8.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1162_7a311dca7708162fcc047282affe85381e5eabb8.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
